src/controller/OpenRobertaController.vala

```diff
@@ -0,0 +1,238 @@
+            config = new KeyFile ();
+
+            bind_property ("available", status_bar_item, "visible", BindingFlags.SYNC_CREATE);
+            bind_property ("available", open_roberta_window.menu, "visible", BindingFlags.SYNC_CREATE);
+
+            notify["available"].connect (() => {
+                if (!available) {
+                    open_roberta_window.info.text =
+                        "Service openrobertalab is not running.";
+                } else {
+                  open_roberta_window.notify_property ("connected");
+                }
+            });
+
+            Bus.watch_name (BusType.SYSTEM, SERVICE_NAME,
+                BusNameWatcherFlags.AUTO_START, () => {
+                    connect_async.begin ((obj, res) => {
+                        try {
+                            connect_async.end (res);
+                            open_roberta_window.loading = false;
+                            available = true;
+                        } catch (IOError err) {
+                            available = false;
+                            warning ("%s", err.message);
+                        }
+                    });
+                }, () => {
+                    open_roberta_window.loading = true;
+                    open_roberta_window.connected = false;
+                    status_bar_item.connected = false;
+                    available = false;
+                    service = null;
+                    // if the service dies (while running code), definitely
+                    // switch back to brickman
+                    switch_to_brickman_screen ();
+                });
+
+            try {
+                config.load_from_file (CONFIG, KeyFileFlags.KEEP_COMMENTS);
+                open_roberta_window.custom_server.label.text =
+                    config.get_string ("Common", "CustomServer");
+                open_roberta_window.selected_server =
+                    config.get_string ("Common", "SelectedServer");
+            } catch (FileError err) {
+                warning ("FileError: %s", err.message);
+            } catch (KeyFileError err) {
+                warning ("KeyFileError: %s", err.message);
+            }
+
+            open_roberta_window.public_server.button.pressed.connect (on_server_connect);
+            open_roberta_window.custom_server.button.pressed.connect (on_server_connect);
+
+            open_roberta_window.config_custom.button.pressed.connect (on_server_edit);
+
+            open_roberta_window.disconnect_server.button.pressed.connect (on_server_disconnect);
+        }
+
+        async void connect_async () throws IOError {
+            service = yield Bus.get_proxy (BusType.SYSTEM, SERVICE_NAME,
+                "/org/openroberta/Lab1");
+            service.status.connect (on_status_changed);
+        }
+
+        void on_status_changed (string message) {
+            debug ("service status: '%s'", message);
+            string[] offline = { "connected", "disconnected" };
```

Comment:
Added comments for the states. Changed it to online containing (registered, executing) and flipped the cond.